views/product_product_views.xml

```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+    <record id="product_prduct_inhert_flower" model="ir.ui.view">
+        <field name="name">product.prduct.inhert.flower</field>
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_normal_form_view"/>
+        <field name="arch" type="xml">
+        
+        
+            <xpath expr="//field[@name='categ_id']" position="after">
+                
+                <field name="needs_watering" invisible="1"/>
+                <field name="flower_id"/>
+                <field name="sequence_id"/>
+                <field name="gardeners" widget="many2many_tags" attrs="{'invisible': [('is_flower', '=', False)]}"/>
+
+            </xpath>
+
+            <xpath expr="//field[@name='purchase_ok']" position="before">
+                <field name="is_flower"/>
+                <label for="is_flower"/>
+            </xpath>
+
+
+        </field>
+    </record>
+
+    <record id="product_flower_mnue_action" model="ir.actions.act_window">
+        <field name="name">product flowers</field>
+        <field name="res_model">product.product</field>
+        <field name="view_mode">kanban,tree,form,activity</field>
+        <field name="context" eval="{'default_is_flower': True}"/>
+        <field name="domain" eval="[('is_flower', '=', True)]"/>
+        
+    </record>
+   
+    <menuitem id="menu_action_flower_product" action="product_flower_mnue_action" parent="sale.product_menu_catalog"
+              sequence="20"/>
+
+
+<record id="sale_order_line_inhert_flower" model="ir.ui.view">
+    <field name="name">sale.order.line.inhert.flower</field>
+    <field name="model">sale.order</field>
+    <field name="inherit_id" ref="sale.view_order_form"/>
+    <field name="arch" type="xml">
+        <xpath expr="//field[@name='order_line']/tree/field[@name='product_id']" position="attributes">
+            <attribute name="domain">[('sale_ok', '=', True), '|', ('company_id', '=', False), ('company_id', '=', parent.company_id), ('is_flower', '=', True)]</attribute>
+        </xpath>
+    </field>  
+</record>
+
+
+            <record id="product_prduct_inhert_flower_ribbon" model="ir.ui.view">
+                <field name="name">product.prduct.inhert.flower</field>
+                <field name="model">product.product</field>
+                <field name="inherit_id" ref="product.product_normal_form_view"/>
+                <field name="arch" type="xml">
+                <xpath expr="//field[@name='image_1920']" position="before">
+                    <widget name="web_ribbon" title="NEEDS WATERING" bg_color="bg-danger" attrs="{'invisible': [('needs_watering', '=', False)]}"/>
+                </xpath>
+                </field>  
+            </record> 
```

Comment:
a preferred way is to use the same inherited view record to do all the XPATHs (refer to line 4)